Docs9159/metrics overview page (#25830)
DataDog/documentation: content/en/metrics/overview.md

Reviewer: Can we remove the new users vs old users elaborate, and change it to something along these lines:
"You can easily explore the sources of their metrics, whether standard or custom, generate additional metrics from Datadog products, and enable advanced configurations like Metrics without Limits and historical data ingestion"?

{{% collapse-content title="Datadog Agent" level="h4" %}}
The [Datadog Agent][3] collects metrics from the hosts it's installed on, and forwards them to Datadog. These metrics could originate from:

   - Any of the official Datadog integrations that are bundled with the Agent. See the [integrations-core repository][4] for a full listing of the available integrations, and [Integration Management][5] for more information about managing these integrations.
   - [DogStatsD][6], a metrics aggregation service bundled with the Datadog Agent. DogStatsD implements the [StatsD][7] protocol with some Datadog-specific extensions.
   - [Custom checks][8], which are used to collect metrics from custom applications or unique systems. You define the logic for the check yourself in the Agent's configuration files. See [Writing a Custom Agent Check][9] for more information.
   - [Marketplace integrations][10] installed on the Agent. The [Datadog Marketplace][11] is a digital marketplace where Technology Partners can list their paid offerings to Datadog users.
{{% /collapse-content %}}

> {{% collapse-content title="Cloud integrations" level="h4" %}}

Reviewer: @buraizu should we capitalize "Integrations"?

Reviewer 2: +1

Author: Unless part of an H1 header, titles are written in sentence case ([style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) for reference)

The sections below detail the different parts of the page and provide guidance on how you can maximize the value of your Datadog metrics.

## How your metrics flow through Datadog

{{< img src="metrics/overview/how_metrics_flow.png" alt="The how your metrics flow through Datadog section of the metrics overview page" >}}

The topmost section of the page shows all of your metric sources, along with any configurable processing and a breakdown of the metrics by type (standard or [custom][1]).

### Metric Sources

The **Metric Sources** column shows a summary of the metric sources reporting to Datadog. Click on any of the sources to open the [Summary page][2] scoped to that source. Your Datadog metrics can originate from the following sources:

{{% collapse-content title="Datadog Agent" level="h4" %}}
The [Datadog Agent][3] collects metrics from the hosts it's installed on, and forwards them to Datadog. These metrics could originate from:

>    - Any of the official Datadog integrations that are bundled with the Agent. See the [integrations-core repository][4] for a full listing of the available integrations, and [Integration Management][5] for more information about managing these integrations.

Reviewer 2: Would be good to include a number like 700+ or 800+ (whatever our current state is)

Author: Thanks for that feedback. We have a way of dynamically rendering the latest number of integrations, but adding it here breaks the formatting of the `collapse-content` section. In [803ca9f](https://github.com/DataDog/documentation/pull/25830/commits/803ca9ff51587e3b1aad595e46e6e064d172caed), I've added this information at the end of the section. Thanks in advance for any additional feedback.

   - [DogStatsD][6], a metrics aggregation service bundled with the Datadog Agent. DogStatsD implements the [StatsD][7] protocol with some Datadog-specific extensions.
   - [Custom checks][8], which are used to collect metrics from custom applications or unique systems. You define the logic for the check yourself in the Agent's configuration files. See [Writing a Custom Agent Check][9] for more information.
   - [Marketplace integrations][10] installed on the Agent. The [Datadog Marketplace][11] is a digital marketplace where Technology Partners can list their paid offerings to Datadog users.
{{% /collapse-content %}}

{{% collapse-content title="Cloud integrations" level="h4" %}}
Also known as authentication-based integrations, these integrations are set up in Datadog. You provide the credentials for obtaining metrics, and Datadog makes API calls on your behalf to collect them. Common examples are cloud provider integrations, Slack, and PagerDuty. See [API-based integrations][12] in the Developers documentation for more information.
{{% /collapse-content %}} 

{{% collapse-content title="Datadog API" level="h4" %}}

> You can send metrics directly to the [Metrics API][13].

Reviewer 2: The current link only brings the user to the list of all APIs - please update the link to be this one https://docs-staging.datadoghq.com/docs9159/metrics-overview-page/api/latest/metrics/#submit-metrics

Author: Updated in [803ca9f](https://github.com/DataDog/documentation/pull/25830/commits/803ca9ff51587e3b1aad595e46e6e064d172caed)

##### Logs

[Generate custom metrics from ingested logs][14] to summarize data from all the logs ingested by Datadog. This allows you to visualize and alert on log data that's important to your environment, even if the logs are not indexed for long-term search.

##### APM

[Generate custom metrics from your ingested spans][15] to visualize anomalies and trends across any parameters that are important to your business context.

##### Real User Monitoring (RUM)

[Generate custom metrics from RUM events][16] to summarize the data from your RUM events, so you can visualize and alert on the user behaviors that are most impactful to your organization.

##### Processes

[Generate custom process-based metrics][17] to monitor the resource consumption of your processes, as well as any other process-related behavior that might be important to your business needs.

##### Events

[Generate custom event-based metrics][18] for visibility into monitor alerts or any other event-based data ingested by Datadog.

{{% /collapse-content %}} 

### Configurable Processing

> The **Configurable Processing** column shows an overview of the configuration options available for your metrics. Click on any of the options for more information and a link to the relevant configuration screen.

Reviewer 2: ```suggestion
The **Configurable Processing** column lists the various advanced configuration options that increase the value of your metrics. Click on any of the options for more information and a link to the relevant configuration screen.
```

{{% collapse-content title="Optimize your custom metrics costs with Metrics without Limits™" level="h4" %}}

> [Metrics without Limits™][19] helps you manage custom metric costs by indexing only the metric tags most valuable to your organization.

Reviewer 2: ```suggestion
[Metrics without Limits™][19] helps you manage custom metric costs by indexing only the metric tags most valuable to your organization. Your use of Metrics without Limits is reflected in the top section of the Overview page as "Selected Metrics". For more best practices on how to manage your custom metrics costs, please refer to our end to end guide here[]
```

Link to governance guide : https://docs.datadoghq.com/metrics/guide/custom_metrics_governance

Author: Link added in [803ca9f](https://github.com/DataDog/documentation/pull/25830/commits/803ca9ff51587e3b1aad595e46e6e064d172caed)

{{% /collapse-content %}} 

{{% collapse-content title="Enable percentiles on distribution metrics" level="h4" %}}

> [Distribution metrics][20] with percentiles enabled show you percentile aggregations of metric values sent from multiple hosts, so you can measure statistical distributions across your entire infrastructure.

Reviewer 2: ```suggestion
[Distribution metrics][20] with percentiles enabled provide you with globally accurate percentiles calculated serverside from all hosts, so you can measure statistical distributions across your entire distributed infrastructure.
```

{{% /collapse-content %}} 

{{% collapse-content title="Ingest historical metrics" level="h4" %}}
[Historical Metrics Ingestion][21] enables you to ingest metric values with timestamps older than one hour from the time of submission.
{{% /collapse-content %}} 

### Available Metrics

The **Available Metrics** column breaks down your total metric volume by standard and custom metrics over the past month.

## Your metrics by source

> This section contains a tree map with a breakdown of your metric sources and their respective volumes.

Reviewer 2: ```suggestion
This section contains a tree map that outlines your metric sources and their respective volumes.
```

{{< img src="metrics/overview/metrics_by_source.png" alt="The your metrics by source section of the metrics overview page" >}}

## Generate metrics from any source

> Click any of the options to open the page in Datadog where you can create custom metrics for:

Reviewer 2: ```suggestion
By clicking on any of the options below, you will be brought to the corresponding product's Generate Metrics page where you can create custom metrics from that respective product: ```

   - [Logs][14]
   - [Ingested spans][15]
   - [RUM events][16]
   - [Processes][17]
   - [Events][18]

## Metrics available for querying

> Use the search bar in this section to view the latest data and configuration options of any of your metrics. You can search all its tags, or click to investigate the metric further in the Metrics Explorer or Summary pages.

Reviewer 2: can we provide the in app links for both the explorer and summary pages here

Author: Added in [803ca9f](https://github.com/DataDog/documentation/pull/25830/commits/803ca9ff51587e3b1aad595e46e6e064d172caed)

---
title: Metrics Overview Page
aliases:
  - /metrics/faq/metrics-without-limits/
  - /metrics/guide/metrics-without-limits-getting-started/
further_reading:
  - link: "https://www.datadoghq.com/blog/metrics-without-limits"
    tag: "Blog"
    text: "Dynamically control custom metrics volume with Metrics without Limits™"
  - link: "https://dtdg.co/fe"
    tag: "Foundation Enablement"
    text: "Join an interactive session to unlock the full potential of metrics"
---

## Overview

The Metrics Overview page provides you with a deeper understanding of your metrics landscape, and is helpful for users of all experience levels. 

>    - **New users** can use its comprehensive overview of Datadog metric offerings to help increase the value of their metrics.

Author: Yep, updated in [428e940](https://github.com/DataDog/documentation/pull/25830/commits/428e940bd1769d53c7c7219fcc217b3410738a63)

   - **Existing users** gain a detailed overview into their current metrics' volumes and origins.

The sections below detail the different parts of the page and provide guidance on how you can maximize the value of your Datadog metrics.

## How your metrics flow through Datadog

> {{< img src="metrics/overview/how_metrics_flow.png" alt="The how your metrics flow through Datadog section of the metrics overview page" >}}

Author: Updated in [e23ed5f](https://github.com/DataDog/documentation/pull/25830/commits/e23ed5f84c5e259c2d812070db70f2d5f275fff2)

> {{< img src="metrics/overview/how_metrics_flow.png" alt="The how your metrics flow through Datadog section of the metrics overview page" >}}

Reviewer: Note that we're making small changes to the top chart of the overview page, so lets make sure to circle back here next week and include the most updated version (should be deployed on Monday/Tuesday next week

> {{< img src="metrics/overview/how_metrics_flow.png" alt="The how your metrics flow through Datadog section of the metrics overview page" >}}

Reviewer: @buraizu the changes have been made.

The topmost section of the page shows all of your metric sources, along with any configurable processing and a breakdown of the metrics by type (standard or [custom][1]).

### Metric Sources

The **Metric Sources** column shows a summary of the metric sources reporting to Datadog. Click on any of the sources to open the [Summary page][27] scoped to that source. Your Datadog metrics can originate from the following sources:

{{% collapse-content title="Datadog Agents" level="h4" %}}
The [Datadog Agent][2] collects metrics from the hosts it's installed on, and forwards them to Datadog. These metrics could originate from:

>    - Any of the official Datadog integrations that are bundled with the Agent. See [Integration Management][3] for more information about managing these integrations, or the [integrations-core repository][4] for a full listing of the available integrations.

Author: Updated in [428e940](https://github.com/DataDog/documentation/pull/25830/commits/428e940bd1769d53c7c7219fcc217b3410738a63)

>    - Any of the official Datadog integrations that are bundled with the Agent. See [Integration Management][3] for more information about managing these integrations, or the [integrations-core repository][4] for a full listing of the available integrations.

Author: Updated in [428e940](https://github.com/DataDog/documentation/pull/25830/commits/428e940bd1769d53c7c7219fcc217b3410738a63)

>    - Any of the official Datadog integrations that are bundled with the Agent. See [Integration Management][3] for more information about managing these integrations, or the [integrations-core repository][4] for a full listing of the available integrations.

Reviewer: Title should be "Datadog Agent" and not "Agents"

>    - Any of the official Datadog integrations that are bundled with the Agent. See [Integration Management][3] for more information about managing these integrations, or the [integrations-core repository][4] for a full listing of the available integrations.

Author: Updated in [c444633](https://github.com/DataDog/documentation/pull/25830/commits/c444633e410dd47d77b84aad0afb84f01de70eaa), but just pointing out this is now inconsistent with what's rendered in the UI

   - [Custom checks][5], which are used to collect metrics from custom applications or unique systems. You define the logic for the check yourself in the Agent's configuration files. See [Writing a Custom Agent Check][6] for more information.
   - [DogStatsD][7], a metrics aggregation service bundled with the Datadog Agent. DogStatsD implements the [StatsD][8] protocol with some Datadog-specific extensions.
   - [Marketplace integrations][9] installed on the Agent. The [Datadog Marketplace][10] is a digital marketplace where Technology Partners can list their paid offerings to Datadog users.
{{% /collapse-content %}}

> {{% collapse-content title="Cloud integrations" level="h4" %}}

Author: Updated in [428e940](https://github.com/DataDog/documentation/pull/25830/commits/428e940bd1769d53c7c7219fcc217b3410738a63)

Also known as authentication-based integrations, these integrations are set up in Datadog. You provide the credentials for obtaining metrics, and Datadog makes API calls on your behalf to collect them. Common examples are cloud provider integrations, Slack, and PagerDuty.
{{% /collapse-content %}} 

{{% collapse-content title="Datadog API" level="h4" %}}
You can send metrics directly to the [Metrics API][11].
{{% /collapse-content %}} 

> {{% collapse-content title="Metrics from other Datadog Services" level="h4" %}}

Author: Updated in [428e940](https://github.com/DataDog/documentation/pull/25830/commits/428e940bd1769d53c7c7219fcc217b3410738a63)

> Some Datadog services, such as Synthetic Testing, provide standard metrics by default. Other services can be configured to generate custom metrics. This section covers the available options and provides links to relevant documentation.

Author: Our [style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#words-and-phrases) requests that we use `service` instead of `product`. With that said, if there's a compelling reason for using `product` instead, I'm fine with that for now. I can follow up with my team later about whether that guideline needs to updated.

> Some Datadog services, such as Synthetic Testing, provide standard metrics by default. Other services can be configured to generate custom metrics. This section covers the available options and provides links to relevant documentation.

Reviewer: Reason would be:
1. To align it with copy on our page
2. There are indeed products which allow you to generate metrics.
I'd push here to change this

##### APM

- [Trace metrics][13] are standard metrics that capture request counts, error counts, and latency measures on your applications instrumented with Datadog APM. You can also [generate custom metrics from your ingested spans][15].
- [Runtime metrics][14] are standard metrics that provide additional insights into an application’s performance.

##### Real User Monitoring (RUM)

- [Web and mobile vitals][16] are standard metrics that provide performance scores for browser and mobile applications, respectively.
- [Custom metrics from RUM events][17] enable you to summarize the data from your RUM events. 

##### Processes

[Process-based metrics][19] are custom metrics you can generate in the [Processes page][25].

##### Events

[Event-based metrics][20] are custom metrics you can generate in the [settings page of Event Management][26].

##### Synthetic Monitoring & Continuous Testing Metrics

[Synthetic Monitoring & Continuous Testing Metrics][18] are standard metrics that are generated to help track test duration, latency, and more.

> ##### Account Management

Author: `synthetics.*` metrics aren't listed as custom metrics in the Plan & Usage page, and I've additionally located some internal resources that verify this. Let me know if there is any additional context around this I'm unaware of, or if you have any feedback.

> ##### Account Management

Reviewer: This might sound a bit confusing, but this section only pertains to product where its possible to generate custom metrics (thus billable), which includes: logs, APM, rum and processes. Can we please fix up this section accordingly?

> ##### Account Management

Author: Thanks for clarifying, I updated this section in [aaedfa1](https://github.com/DataDog/documentation/pull/25830/commits/aaedfa1dd0509ef6c265a3c89d308ae6faa5dcc5) to:
- Remove anything about standard metrics
- Emphasize the use cases for creating the different custom metrics

> ##### Account Management

Reviewer: I see that this still has products which aren't logs, APM, rum and processes. @buraizu any chance we can remove this?

[Estimated Usage Metrics][21] are standard metrics that Datadog calculates to provide estimates of your current usage.
{{% /collapse-content %}} 

### Configurable Processing

The **Configurable Processing** column shows an overview of the configuration options available for your metrics. Click on any of the options for more information and a link to the relevant configuration screen.

> {{% collapse-content title="Manage tagging" level="h4" %}}

Author: This section does link to the metrics without limits page, but I've updated the text in [922a9cf](https://github.com/DataDog/documentation/pull/25830/commits/922a9cfb6094844611d96d6278fcf030fb6c9abf) to make that more explicit.

> {{% collapse-content title="Manage tagging" level="h4" %}}

Reviewer: Lets change this title to "Optimize your custom metrics costs with Metrics without Limits"

[Configuration of tags][22] can help you to manage custom metric costs by indexing only the metric tags most valuable to your organization.
{{% /collapse-content %}} 

{{% collapse-content title="Enable percentiles" level="h4" %}}

> [Distributions][23] show you percentile aggregations of metric values sent from multiple hosts, so you can measure statistical distributions across your entire infrastructure.

Author: Thanks for that feedback, updated in [922a9cf](https://github.com/DataDog/documentation/pull/25830/commits/922a9cfb6094844611d96d6278fcf030fb6c9abf)

{{% /collapse-content %}} 

{{% collapse-content title="Ingest historical metrics" level="h4" %}}
[Historical Metrics Ingestion][24] enables you to ingest metric values with timestamps older than one hour from the time of submission.
{{% /collapse-content %}} 

### Available Metrics

> 

Author: Would these be separate products from those already listed in the [metric sources section](https://docs-staging.datadoghq.com/docs9159/metrics-overview-page/metrics/overview/#metric-sources)?

> 

Reviewer: True, just wondering from a doc perspective if this is clear (to skip one cell). I would think maybe to link to the first one? Whats typically done in these situations?

> 

Author: Thanks for clarifying, I see what you mean and added a section in [f6b57dd](https://github.com/DataDog/documentation/pull/25830/commits/f6b57dd6180c2507ef5c1f85c312487b8ad7df33)

   - [Custom checks][8], which are used to collect metrics from custom applications or unique systems. You define the logic for the check yourself in the Agent's configuration files. See [Writing a Custom Agent Check][9] for more information.
   - [Marketplace integrations][10] installed on the Agent. The [Datadog Marketplace][11] is a digital marketplace where Technology Partners can list their paid offerings to Datadog users.

{{% /collapse-content %}}

{{% collapse-content title="Cloud integrations" level="h4" %}}
Also known as authentication-based integrations, these integrations are set up in Datadog. You provide the credentials for obtaining metrics, and Datadog makes API calls on your behalf to collect them. Common examples are cloud provider integrations, Slack, and PagerDuty. See [API-based integrations][12] in the Developers documentation for more information.
{{% /collapse-content %}} 

{{% collapse-content title="Datadog API" level="h4" %}}
You can send metrics directly to the [Metrics API][13].
{{% /collapse-content %}} 

{{% collapse-content title="Metrics from other Datadog products" level="h4" %}}

> Some Datadog products, such as Synthetic Testing, provide standard metrics by default. Other products can be configured to generate custom metrics. This section covers the available options and provides links to relevant documentation.

Author: Thanks for the feedback, updated in [1b44ba0](https://github.com/DataDog/documentation/pull/25830/commits/1b44ba00993ee1782a01a78955520a81546dc5d9)

### Configurable Processing

The **Configurable Processing** column lists the various advanced configuration options that you can use to increase the value of your metrics. Click on any of the options for more information and a link to the relevant configuration screen.

{{% collapse-content title="Optimize your custom metrics costs with Metrics without Limits™" level="h4" %}}
[Metrics without Limits™][19] helps you manage custom metric costs by indexing only the metric tags most valuable to your organization. Your use of Metrics without Limits™ is reflected in the top section of the Overview page as **Selected Metrics**. For more information about managing your custom metrics costs, see [Best Practices for Custom Metrics Governance][22].
{{% /collapse-content %}} 

{{% collapse-content title="Enable percentiles on distribution metrics" level="h4" %}}
[Distribution metrics][20] with percentiles enabled provide you with globally accurate percentiles calculated server-side from all hosts, so you can measure statistical distributions across your entire distributed infrastructure.
{{% /collapse-content %}} 

> {{% collapse-content title="Metrics from other Datadog products" level="h4" %}}

Reviewer 2: ```suggestion
{{% collapse-content title="Generate metrics from other Datadog products" level="h4" %}}
```

##### APM

[Generate custom metrics from your ingested spans][15] to visualize anomalies and trends across any parameters that are important to your business context.

##### Real User Monitoring (RUM)

[Generate custom metrics from RUM events][16] to summarize the data from your RUM events, so you can visualize and alert on the user behaviors that are most impactful to your organization.

##### Processes

[Generate custom process-based metrics][17] to monitor the resource consumption of your processes, as well as any other process-related behavior that might be important to your business needs.

##### Events

[Generate custom event-based metrics][18] for visibility into monitor alerts or any other event-based data ingested by Datadog.
{{% /collapse-content %}} 

{{% collapse-content title="Ingest historical metrics" level="h4" %}}
[Historical Metrics Ingestion][21] enables you to ingest metric values with timestamps older than one hour from the time of submission.
{{% /collapse-content %}} 

### Available Metrics

> The **Available Metrics** column breaks down your total metric volume by standard and custom metrics over the past month. If you're interested in managing your custom metrics volume, see the [Best Practices for Custom Metrics Governance page][25] and the [metrics volume management page][26] for more detailed insights.

Reviewer 2: ```suggestion
The **Available Metrics** column breaks down your total metric volume by standard and custom metrics over the past month. If you're interested in managing your custom metrics volume, see the [Best Practices for Custom Metrics Governance page][25] and the [Metrics Volume Management page][26] for more detailed insights.
```

Reviewer 2: All Metrics pages - Metrics Explorer, Metrics Overview, Metrics Summary have been capitalized in most existing metrics doc that I've written previously... (Metrics Volume is shorthand for Metrics Volume Management and is called out in our volume docs 
<img width="917" alt="Screenshot 2024-11-13 at 12 35 16 PM" src="https://github.com/user-attachments/assets/7f109357-0539-4fa7-af88-d1c6711d24b5">

## Your metrics by source

This section contains a tree map that outlines your metric sources and their respective volumes.

{{< img src="metrics/overview/metrics_by_source.png" alt="The your metrics by source section of the metrics overview page" >}}

## Generate metrics from any source

Clicking on any of the options below brings you to the corresponding product's Generate Metrics page, where you can create custom metrics from that product:

>    - [Ingested Logs][14]

Reviewer 2: ```suggestion
   - [Ingested logs][14]
```
